statistics/table.go

```diff
@@ -687,3 +689,33 @@ func (coll *HistColl) GetAvgRowSize(cols []*expression.Column, isEncodedKey bool
 	// Add 1 byte for each column's flag byte. See `encode` for details.
 	return size + float64(len(cols))
 }
+
+// GetTableAvgRowSize computes average row size for a table scan, exclude the index key-value pairs.
+func (coll *HistColl) GetTableAvgRowSize(cols []*expression.Column, storeType kv.StoreType, pkIsHandle bool) (size float64) {
+	size = coll.GetAvgRowSize(cols, false)
+	switch storeType {
+	case kv.TiKV:
+		size += tablecodec.RecordRowKeyLen
+		if pkIsHandle {
```

Comment:
I just recalled that, in `buildDataSource`, we would append `_tidb_rowid` into `TblCols`, so even if `pkIsHandle` is false, we should do `size -= 8` here? Also we can remove the below TiFlash branch?

Reply:
The columns of TiFlash is `tableScan.schema.Columns`. I think it can't be removed.